src/main.c

```diff
@@ -86,6 +86,13 @@ int main(int argc, char **argv)
 		// this launcher
 		setup_slave_mount_namespace();
 
+		// Get the current working directory before we start fiddling with
+		// mounts and possibly pivot_root.  At the end of the whole process, we
+		// will try to re-locate to the same directory (if possible).
+		char vanilla_cwd[512];
+		if (getcwd(vanilla_cwd, sizeof vanilla_cwd) == NULL) {
+			die("cannot get the current working directory");
+		}
```

Comment:
I suggest using get_current_dir_name() instead of getcwd(), like we do in mount-support.c. This has the advantage of not having to pick an arbitrary size for vanilla_cwd but does mean you need to free() the result.